File: dart_native/example/lib/android/constructorTest.dart
import 'dart:ffi';

import 'package:dart_native/dart_native.dart';
import 'package:dart_native_example/android/entity.dart';

class constructorTest extends JObject {
  constructorTest(String s, [Pointer ptr])
      : super("com/dartnative/dart_native_example/constructorTest", ptr, [s]);

  String getString() {
    return invoke('getString', [], "Ljava/lang/String;", "");
  }

  void testStatic(){
    return invoke('testStatic', [], "V", "static");
  }

  Entity createEntity(){
    return new Entity(invoke('createEntity', [], 'Lcom/dartnative/dart_native_example/Entity;',""));
  }

}

class InnerC extends JObject {
  InnerC([Pointer ptr]) : super("com/dartnative/dart_native_example/constructorTest\$InnerC", ptr, []);

  void testInnerC() {
    return invoke('testInnerC', [], 'V', "static");
  }
}

File: dart_native/example/android/app/src/main/java/com/dartnative/dart_native_example/constructorTest.java
package com.dartnative.dart_native_example;

import io.flutter.Log;

/**
 * Created by huizzzhou on 2020/6/4.
 */
public class constructorTest {

    private static final String TAG = "dart_java";

    private String name = "none";

    int[] intArray = new int[]{5,2,3,4};

    constructorTest(String str){
        Log.d(TAG, this.name);
        this.name = str;
        Log.d(TAG, this.name);
    }

    public String getString(){
        return name;
    }

    public static void testStatic(){
        Log.d(TAG, "testStatic");
    }

    public Entity createEntity(){
        return new Entity();
    }

}
